msal/src/Microsoft.Identity.Client/UIParent.cs

```diff
@@ -73,8 +74,11 @@ public UIParent(bool useEmbeddedWebview) : this()
 #if ANDROID
         private Activity Activity { get; set; }
 
-        private static readonly string[] _chromePackages =
-        {"com.android.chrome", "com.chrome.beta", "com.chrome.dev"};
+        private static readonly string ChromePackage = "com.android.chrome";
+
+        private static readonly string[] _customTabPackages =
+        { ChromePackage, "com.chrome.beta", "com.chrome.dev", "com.microsoft.emmx", "org.mozilla.firefox",
+        "com.ecosia.android", "com.kiwibrowser.browser", "com.brave.browser"};
```

Comment:
can't we ask the browser itself if it supports custom tabs? Do we really need to maintain this list out selves?

Reply:
i'll look into it.

---
In reply to: [227919485](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1347#discussion_r227919485) [](ancestors = 227919485)

Reply:
great. I'll approve this PR and you can investigate and add that particular change in a later PR. Thanks!

---
In reply to: [227920771](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1347#discussion_r227920771) [](ancestors = 227920771,227919485)

Reply:
ok...pushed up some changes. lmk what you think.

---
In reply to: [227919485](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1347#discussion_r227919485) [](ancestors = 227919485)

Reply:
Looks much better to me :)

---
In reply to: [227935640](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1347#discussion_r227935640) [](ancestors = 227935640,227919485)